cpp/src/strings/repeat_strings.cu

```diff
@@ -235,6 +235,8 @@ struct compute_size_and_repeat_separately_fn {
  * This function is similar to `strings::detail::make_strings_children`, except that it accepts an
  * optional input `std::optional<column_view>` that can contain the precomputed sizes of the output
  * strings.
+ *
+ * @deprecated This will be removed with issue 12542
```

Comment:
```suggestion
 * @deprecated The parameter `output_strings_sizes` will be removed with issue 12542
```

Reply:
This whole function can be removed as part of 12542